package/MDAnalysis/analysis/encore/confdistmatrix.py

```diff
@@ -0,0 +1,435 @@
+
+        # Initialize workers. Simple worker doesn't perform fitting,
+        # fitter worker does.
+        if pairwise_align:
+            workers = [Process(target=self._fitter_worker, args=(
+                tasks_per_worker[i],
+                ensemble.trajectory.timeseries(
+                    ensemble.select_atoms(selection),
+                    format='fac'),
+                ensemble.trajectory.timeseries(
+                    ensemble.select_atoms(subset_selection),
+                    format='fac'),
+                masses,
+                subset_masses,
+                distmat,
+                partial_counters[i])) for i in range(ncores)]
+        else:
+            workers = [Process(target=self._simple_worker,
+                               args=(tasks_per_worker[i],
+                                     ensemble.trajectory.timeseries(
+                                         ensemble.select_atoms(selection),
+                                         format='fac'),
+                                     masses, distmat,
+                                     partial_counters[i]))
+                       for i in range(ncores)]
+
+        workers += [Process(target=self._pbar_updater,
+                            args=(pbar, partial_counters, matsize))]
+
+        # Start & join the workers
+        for w in workers:
+            w.start()
+        for w in workers:
+            w.join()
+
+        # When the workers have finished, return a TriangularMatrix object
+        return TriangularMatrix(distmat, metadata=metadata)
+
+    @staticmethod
+    def _simple_worker():
+        '''Simple worker prototype; to be overriden in derived classes
+        '''
+        return None
+
+    @staticmethod
+    def _fitter_worker():
+        """
+        Fitter worker prototype; to be overridden in derived classes
+        """
+        return None
+
+    @staticmethod
+    def _pbar_updater(pbar, pbar_counters, max_val, update_interval=0.2):
+        '''Method that updates and prints the progress bar, upon polling
+        progress status from workers.
+
+        Attributes
+        -----------
+
+        pbar : encore.utils.AnimatedProgressBar object
+            Progress bar object
+
+        pbar_counters : list of multiprocessing.RawValue
+            List of counters. Each worker is given a counter, which is updated
+            at every cycle. In this way the _pbar_updater process can
+            asynchronously fetch progress reports.
+
+            max_val : int
+            Total number of matrix elements to be calculated
+
+            update_interval : float
+            Number of seconds between progress bar updates
+
+        '''
+
+        val = 0
+        while val < max_val:
+            val = 0
+            for c in pbar_counters:
+                val += c.value
+            pbar.update(val)
+            pbar.show_progress()
+            sleep(update_interval)
+
+    __call__ = run
```

Comment:
This makes me realize that this shouldn't be a class. Besides `run` everything is static and even `run` doesn't use the `self` variable. You should refactor this into individual functions.